File: PYTHON/countmax.py
def countMax(upRight):
    rows, cols = 0, 0

    # Process each coordinate and determine the final size of the grid
    for coord in upRight:
        r, c = map(int, coord.split())
        rows = max(rows, r)
        cols = max(cols, c)

    # Create the grid and fill it with zeros
    grid = [[0] * (cols + 1) for _ in range(rows + 1)]
    print (grid)

    # Update the grid based on the coordinates
    for coord in upRight:
        r, c = map(int, coord.split())
        for i in range(1, r + 1):
            for j in range(1, c + 1):
                grid[i][j] += 1

    # Find the maximal element in the grid
    max_element = max(max(row) for row in grid)

    # Count occurrences of the maximal element
    count_max = sum(row.count(max_element) for row in grid)

    return count_max

# Sample Input
upRight = ["3 2", "3 7", "4 1"]

# Sample Output
result = countMax(upRight)
print(result)

File: CPP/countmax.cpp
#include<iostream>
#include<vector>
#include<sstream>
#include<algorithm>

using namespace std;

long countMax(vector<string> upRight) {
    int rows = 0, cols = 0;

    for (string coord : upRight) {
        istringstream iss(coord);
        int r, c;
        iss >> r >> c;
        rows = max(rows, r);
        cols = max(cols, c);
    }

    vector<vector<int>> grid(rows + 1, vector<int>(cols + 1, 0));

    for (string coord : upRight) {
        istringstream iss(coord);
        int r, c;
        iss >> r >> c;
        for (int i = 1; i <= r; i++) {
            for (int j = 1; j <= c; j++) {
                grid[i][j] += 1;
                //cout<<grid[i][j]<<"\t";
            }
            //cout<<endl;
        }
        //cout<<endl;
    }

    int max_elem = 0;
    int my_elem = 0;
    for (vector<int> row : grid) {
        my_elem = *max_element(row.begin(), row.end());
        if (my_elem > max_elem)
            max_elem = my_elem;
    }
    cout<<"Max Elements: "<<max_elem<<endl;

    int count_max = 0;
    for (vector<int> row : grid) {
        count_max += count(row.begin(), row.end(), max_elem);
    }


    return count_max;
}

int main() {
    vector<string> upRight = {"1 4", "2 3", "4 1"};
    int result = countMax(upRight);
    cout <<"Count Occurences: "<< result << endl;
    return 0;
}
